service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java

```diff
@@ -262,4 +259,24 @@ private String buildNotePertinentInformation(AllergyIntolerance allergyIntoleran
             .map(Annotation::getText)
             .collect(Collectors.joining(StringUtils.SPACE));
     }
+
+    private void buildAuthor(AllergyIntolerance allergyIntolerance, AllergyStructureTemplateParametersBuilder templateParameter) {
+        if (isValidAsserter(allergyIntolerance)) {
+            buildParticipant(allergyIntolerance.getAsserter(), ParticipantType.PERFORMER)
+                .ifPresent(templateParameter::performer);
+        } else if (allergyIntolerance.hasRecorder()) {
+            buildParticipant(allergyIntolerance.getRecorder(), ParticipantType.PERFORMER)
+                .ifPresent(templateParameter::performer);
+        }
+    }
+
+    private Boolean isValidAsserter(AllergyIntolerance allergyIntolerance) {
+        if (allergyIntolerance.hasAsserter()) {
```

Comment:
I would reduce it to a simple expression:
```java
private static boolean isValidAsserter(AllergyIntolerance allergyIntolerance) {
	return allergyIntolerance.hasAsserter() && allergyIntolerance.getAsserter().getReferenceElement().getResourceType().startsWith(ResourceType.Practitioner.name());
}
```

Reply:
Fixed this time :D